main.tf

```diff
@@ -370,6 +370,32 @@ resource "aws_iam_role" "instance" {
   tags = merge(local.tags, var.role_tags)
 }
 
+################################################################################
+### Policy for the instance to use the KMS key
+################################################################################
+resource "aws_iam_policy" "instance_kms_policy" {
+  count = var.enable_kms ? 1 : 0
+
+  name        = "${local.name_iam_objects}-kms"
+  path        = "/"
+  description = "Allow runner instance the ability to use the KMS key."
+  policy = templatefile("${path.module}/policies/instance-kms-policy.json",
+    {
+      kms_key_arn = var.enable_kms && var.kms_key_id == "" ? aws_kms_key.default[0].arn : "arn:aws:kms:${var.aws_region}:${data.aws_caller_identity.current.account_id}:key/${var.kms_key_id}"
```

Comment:
I doubt this is correct. On other places in the module var.kms_id is pointing to the ARN as far I can see.

Reply:
Fixed this. var.kms_key_id is now assumed to be the full ARN. Checked a few places where var.kms_key_id is used and they accept the ARN. Some accept the ARN or the key ID.

Reply:
Yeah, horrible in the provider. I updated the variable description with this finding.